platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,387 @@
+        if (distance >= minimumDistance && distance <= maximumDistance) {
+            [validRows addObject:row];
+        }
+    }
+    
+    return validRows;
+}
+
+// Find the row closest to -[MGLScaleBarView preferredWidth]
+- (NSArray<NSNumber *> *)preferredRowInRows:(NSArray<NSArray<NSNumber *> *> *)rows {
+    CLLocationDistance diff = CGFLOAT_MAX;
+    CLLocationDistance preferredDistance = [self preferredWidth] * [self unitsPerPoint];
+    NSArray<NSNumber *> *preferredRow = rows.firstObject;
+    for (NSArray<NSNumber *> *row in rows) {
+        CLLocationDistance currentDiff = ABS(preferredDistance - row.firstObject.floatValue);
+        if (currentDiff < diff) {
+            preferredRow = row;
+        }
+    }
+    return preferredRow;
+}
+
+- (BOOL)usesMetricSystem {
+    NSLocale *locale = [NSLocale currentLocale];
+    return [[locale objectForKey:NSLocaleUsesMetricSystem] boolValue];
+}
+
+- (void)fadeIn {
+    if (self.isVisible) {
+        return;
+    }
+
+    self.visible = YES;
+    
+    [UIView animateWithDuration:.2 delay:0 options:UIViewAnimationOptionBeginFromCurrentState animations:^{
+        self.alpha = 1.0;
+    } completion:nil];
+}
+
+- (void)fadeOut {
+    if (!self.isVisible) {
+        return;
+    }
+    
+    self.visible = NO;
+    
+    [UIView animateWithDuration:.3 delay:0.7 options:UIViewAnimationOptionBeginFromCurrentState animations:^{
+        self.alpha = 0.0;
+    } completion:nil];
+}
+
+#pragma mark - Setters
+
+- (void)setMetersPerPoint:(CLLocationDistance)metersPerPoint {
+    if (_metersPerPoint == metersPerPoint) {
+        return;
+    }
+    
+    [self fadeIn];
+    
+    _metersPerPoint = metersPerPoint;
+    self.row = [self preferredRowInRows:[self validRows]];
+    
+    [self invalidateIntrinsicContentSize];
+    [self setNeedsLayout];
+}
+
+- (void)setRow:(NSArray<NSNumber *> *)row {
+    if (![row isEqualToArray:_row]) {
+        [_bars makeObjectsPerformSelector:@selector(removeFromSuperview)];
+        [_labels makeObjectsPerformSelector:@selector(removeFromSuperview)];
+        _bars = nil;
+        _labels = nil;
+        _row = row;
+    }
+}
+
+#pragma mark - Views
+
+- (NSArray<UIView *> *)bars {
+    if (!_bars) {
+        NSMutableArray *bars = [NSMutableArray array];
+        for (NSUInteger i = 0; i < self.row.lastObject.integerValue; i++) {
+            UIView *bar = [[UIView alloc] init];
+            [bars addObject:bar];
+            [self.containerView addSubview:bar];
+        }
+        _bars = bars;
+    }
+    return _bars;
+}
+
+- (NSArray<UILabel *> *)labels {
+    if (!_labels) {
+        NSMutableArray *labels = [NSMutableArray array];
+        for (NSUInteger i = 0; i <= self.row.lastObject.integerValue; i++) {
+            UILabel *label = [[MGLScaleBarLabel alloc] init];
+            label.font = [UIFont systemFontOfSize:8 weight:UIFontWeightMedium];
+            NSCharacterSet *characterSet = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
```

Comment:
Take this out of the `for` loop. Although `+[NSCharacterSet decimalDigitCharacterSet]` is probably static, the inverted copy of it is not.